initial version with openshift support

djkormo opened the pull request:
<!--

* https://github.com/helm/helm/tree/master/docs/chart_best_practices

Following best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] DCO signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md

Jonsy13 committed:
Added support for proxy envs in `litmus` chart (#390)

* Added support for proxy envs

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>

* Added support for proxy envs

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>

* Updated docs version

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>

---------

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>
Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

Jonsy13 committed:
Updated helm version to 3.18.x (#436)

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>
Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
initial version with openshift support

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
adding version 3.18.0

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

wouter2397 commented on charts/litmus/templates/frontend-route.yaml:
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.openshift.route.enabled -}}
+{{- if and .Values.openshift.route.enabled (.Capabilities.APIVersions.Has "route.openshift.io/v1") }}
```
I've seen in the past that (.Capabilities.APIVersions.Has "route.openshift.io/v1") is not always working right. Sometimes Helm cannot correctly determine the capabilities and there for marks it as false. But leave it in and I will test it.

wouter2397 commented on values-local-3-16-0.yaml:
```diff
@@ -0,0 +1,393 @@
+        targetCPUUtilizationPercentage: 50
+        targetMemoryUtilizationPercentage: 50
+      securityContext:
+        runAsUser: 2000
```
I have removed the runAsUser now from my deployments and they are starting with the arbitrary uids. I need to test further if this is causing my slowness. But would be great if litmus supports the arbitrary uids of OpenShift instead of a static uid

wouter2397 commented on values-local-3-18-0.yaml:
```diff
@@ -0,0 +1,391 @@
+
+# -- Configure the Bitnami MongoDB subchart
+# see values at https://github.com/bitnami/charts/blob/master/bitnami/mongodb/values.yaml
+mongodb:
```
The mongodb dependency chart is not working for me on OpenShift. The first pod keeps in ready 0/1 state which blocks the others from deploying. Have you seen this before in other environments?

djkormo committed:
removing annotations

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
corrrecting serviceaccunt name and user ids

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
correcting local values for 3.18.0 version

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
adding pvc for scc

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
fixing scc permissions

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
adding route host

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
adding default account to scc

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

ispeakc0de committed:
chore(3.19.0): Adding the helm charts for 3.19.0 version (#437)

Signed-off-by: Shubham Chaudhary <shubham.chaudhary@harness.io>
Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

Jonsy13 committed:
Added changes for helm-litmus-3.19.x (#438)

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>
Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
Refactoring and adding 3.19.0 version

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

djkormo committed:
resolving conflicts for 3.19.0 version

Signed-off-by: Krzysztof Pudłowski <djkormo@gmail.com>

mmkay committed:
fix: Use bitnami/os-shell instead of removed bitnami-shell in mongo (#443)

Signed-off-by: Mateusz Kulewicz <mateusz.kulewicz@gmail.com>

Jonsy13 committed:
Added changes for 3.20.x release (#448)

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>

ispeakc0de committed:
release(3.20.0): Add the helm charts for the 3.20.0 (#447)

Signed-off-by: Shubham Chaudhary <shubham.chaudhary@harness.io>
Co-authored-by: Vedant Shrotria <vedant.shrotria@harness.io>

djkormo committed:
resolving conflicts

djkormo committed:
resolving conflict

ispeakc0de committed:
release(3.21.0): Add the helm charts for the 3.21.0 (#453)

Signed-off-by: Shubham Chaudhary <shubham.chaudhary@harness.io>

djkormo committed:
resolving conflicts

Jonsy13 committed:
Added changes for 3.21.x release (#454)

* Added changes for 3.21.x release

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>

* Added changes for 3.21.x release

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>

---------

Signed-off-by: Jonsy13 <vedant.shrotria@harness.io>

djkormo committed:
resolving conflicts for 3.21.0 version

Jonsy13 committed:
Updating MongoDB repository to bitnamilegacy (#457)

* Added fixes

Signed-off-by: git <harness_commons_bot@service.harness.io>

* Added fixes

Signed-off-by: git <harness_commons_bot@service.harness.io>

---------

Signed-off-by: git <harness_commons_bot@service.harness.io>
Co-authored-by: git <harness_commons_bot@service.harness.io>

KelvinVenancio committed:
fix(litmus): add HPA API version compatibility for multiple k8s versions (#455)

* fix(litmus): add HPA API version compatibility for multiple k8s versions

  Add dynamic HPA apiVersion detection to support both older and newer
  Kubernetes versions. The implementation checks for autoscaling/v2
  availability (K8s 1.23+) and falls back to autoscaling/v2beta2 for
  older clusters.

  Changes:
  - Add litmus-portal.hpa.apiVersion helper in _helpers.tpl
  - Update frontend-hpa.yaml to use dynamic apiVersion
  - Update auth-server-hpa.yaml to use dynamic apiVersion

  This ensures compatibility across Kubernetes versions while maintaining
  the existing metrics specification format.

Signed-off-by: Kelvin V. Soares <kelvinvenanciosoares@gmail.com>

* Updating MongoDB repository to bitnamilegacy (#457)

* Added fixes

Signed-off-by: git <harness_commons_bot@service.harness.io>

* Added fixes

Signed-off-by: git <harness_commons_bot@service.harness.io>

---------

Signed-off-by: git <harness_commons_bot@service.harness.io>
Co-authored-by: git <harness_commons_bot@service.harness.io>
Signed-off-by: Kelvin V. Soares <kelvinvenanciosoares@gmail.com>

---------

Signed-off-by: Kelvin V. Soares <kelvinvenanciosoares@gmail.com>
Signed-off-by: git <harness_commons_bot@service.harness.io>
Co-authored-by: Vedant Shrotria <vedant.shrotria@harness.io>
Co-authored-by: git <harness_commons_bot@service.harness.io>

arunkumar-1992 committed:
feat: Add option to disable pre-install hook job via enablePreInstallJob flag (#456)

chore: bump the chart version to 3.21.1




feat(litmus-agent): add enablePreInstallJob flag to disable pre-install hook job and update docs

Signed-off-by: arunkumar-1992 <111547359+arunkumar-1992@users.noreply.github.com>
Co-authored-by: Arunkumar Chithaluri <arunkumar.chithaluri@swisscom.com>

Jonsy13 committed:
Added fixes (#458)

Signed-off-by: git <harness_commons_bot@service.harness.io>
Co-authored-by: git <harness_commons_bot@service.harness.io>

Jonsy13 committed:
Added fixes (#459)

Signed-off-by: git <harness_commons_bot@service.harness.io>
Co-authored-by: git <harness_commons_bot@service.harness.io>

ispeakc0de committed:
chore(3.22.0): Add the helm charts for 3.22.0 version (#460)

* chore(3.22.0): Add the helm charts for 3.22.0 version

Signed-off-by: Shubham Chaudhary <shubham.chaudhary@harness.io>

* update the helm docs

Signed-off-by: Shubham Chaudhary <shubham.chaudhary@harness.io>

* update the helm docs

Signed-off-by: Shubham Chaudhary <shubham.chaudhary@harness.io>

---------

Signed-off-by: Shubham Chaudhary <shubham.chaudhary@harness.io>

Jonsy13 committed:
Merge branch 'master' into litmus-helm-for-openshift
